training/src/main/scala/com/airbnb/aerosolve/training/pipeline/GenericPipeline.scala

```diff
@@ -139,6 +139,23 @@ object GenericPipeline {
     val metrics = evalCompute(sc, config, cfgKey, isTrainingFunc)
 
     metrics.foreach(x => log.info(if (x._1 contains "THRESHOLD") x._1 else x.toString))
+
+    // Write metrics to result
+    try {
```

Comment:
agreed. Could be something like this:

def outputResults(records to write): Unit = {
  // log
  records.map(e => log.info(e.toString))
  // Write to hdfs
  Your logic
}